### DISCUSSION TITLE
[Question] Can i adjust IPF Corner by manually?

### DISCUSSION THREAD

**Discussion by labnewman** [mtex-toolbox/mtex#1361] [Ask Anything]
Hi community,

Sorry for new blood in coding Mtex.
I'm first time to type a code when using Mtex, so it's very hard for me.

question:
Can I change the IPFkey corner from [(0,0,1),(0,1,1),(-1,1,1)], like fig.1, to [(0,0,1),(1,0,1),(111)], like fig.2?
I know these two corner are hkl family. But the latter corner is more common than former one.
And the hkl read out of the color will be different.

already know:
I have searched the same question on the forum.
But I need to actually change the corner rather than just annotate the [(0,0,1),(1,0,1),(111)] text.

Would anyone can give me the Mtex coding to change it?

Best regards!

fig.1
![fig 2](https://user-images.githubusercontent.com/100294586/155339857-8b3bf4f0-ed6a-4aa5-9732-b694e5e336c9.png)
fig.2
![fig 1](https://user-images.githubusercontent.com/100294586/155339850-060c2eba-cdb1-497c-94f2-0b9b147d1101.png)

**Comment by kilir** [comment]
Hi,
see https://github.com/mtex-toolbox/mtex/discussions/1329
Cheers,
Rüdiger

**Reply by labnewman** [reply]
hi killir,

I did see #1329, and tried before. But it seems like that the code just change the text of the corner by annotation method, and the hkl inside the picture is still the same. 
I mean, now, for example, my hkl is (-5,10,12). Do we have a method to change the (-5,10,12) to positive number by changing the corner to  [(0,0,1),(1,0,1),(111)]?

now, my code is:
plotIPDF(big_grains('copper').meanOrientation,colors,vector3d.Z,...
  'MarkerSize',0.05*big_grains('copper').area,'markerEdgeColor','k')

Best regards.
![fig 3](https://user-images.githubusercontent.com/100294586/155442731-a8bd68a9-6ca2-429a-8cf1-336451afd18b.png)

**Reply by kilir** [reply]
I think you did not plot into the right figure.
See here:
```
cs = crystalSymmetry('cubic')
h= Miller({0 0 1},{1 0 1},{1 1 1},cs)

hi = Miller(-5,10,12,cs)

f= sphericalRegion.byVertices(h)
plot(f)
hold on
plot(hi.symmetrise('unique'),'labeled')
hold off
annotate(h.symmetrise('unique'),'labeled')
```

**Reply by labnewman** [reply]
Yes, in this way, it will change into 001, 101, 111 corners. But the number -5,10,12 still doesn't change into combination of 001,101,111.
Can we let the hkl change into combination of 001,101,111? 

thanks!

**Comment by kilir** [comment]
That's what I get using the lines above:
<img width="480" alt="ftrrr2" src="https://user-images.githubusercontent.com/11697557/162678652-46330fe1-21fb-4ca0-8b2a-a9cd42edd4ed.png">

**Reply by labnewman** [reply]
got it! so the hkl is (10,5,12) now! 

by the way, can we change the deviation tolerance? I mean, producer which sells (110) will have statement like this '(101 )+/- 2 degree'.
so i want to know, in my picture, does (102) may become (101) after changing some euler angle torerance? 

thanks! you help me alot!

**Reply by kilir** [reply]
You are welcome.
With respect to your tolerance question, could you be more precise and possibly open a new thread. I never bought a Miller index so far.
Cheers,
Rüdiger
